gnomad/assessment/validity_checks.py

```diff
@@ -1211,3 +1211,80 @@ def count_vep_annotated_variants_per_interval(
     )
 
     return interval_ht
+
+
+def check_missingness_of_struct(
+    struct_expr: hl.expr.StructExpression, prefix: str = ""
+) -> dict:
+    """
+    Recursively check the fraction of missing values of all fields within a StructExpression.
```

Comment:
is this faster than running `.flatten()` and then checking for missingness on the flattened struct?

Reply:
I also just noticed this: https://hail.is/docs/0.2/hail.expr.StructExpression.html#hail.expr.StructExpression.summarize -- do you know if this is too slow to run at scale? I guess it might be annoying to read through too

Reply:
yeah the summarize adds a lot of other things that we don't need

and not sure if this is any faster, i could just flatten them as well and then check missingness